app/controllers/application_controller.rb

```diff
@@ -1,10 +1,79 @@
+require 'pry'
 require './config/environment'
 
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "evelynwaugh"
   end
 
+  helpers do
+    def logged_in?
+      if session[:user_id]
+        true
+      else
+        false
+      end
+    end
+
+  end
+
+  get '/' do
+    erb :'home/index'
+  end
+
+  get '/login' do
+    if logged_in?
+      redirect to '/tweets'
+    end
+    erb :'home/login'
+  end
+
+  get '/signup' do
+    if logged_in?
+      redirect to '/tweets'
+    end
+    erb :'home/signup'
+  end
+
+  post '/signup' do
+    if params[:user][:username].empty? || params[:user][:password].empty? || params[:user][:email].empty?
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.